### PR TITLE
[348] [178] Rework suite assignment for draws

### LINKED ISSUE #178: Add "Check-All" button to Update Suites on Draws
Blocked on #175 

The form that updates suites in draws should include a check all button

### LINKED ISSUE #348: Rework suite assignment for draws
One page / form, split up by suite size (convert `suites_edit.html.erb` to a partial), with check/uncheck all (woo JS, see #178 for details).

### PR DESCRIPTION
**Pull request by orenyk**
Resolves #348, resolves #178
- Consolidate all suite sizes onto one form
- Add method_missing to DrawSuiteUpdate to set dynamic attributes for each suite size
- Update DrawsController to consolidate the params from the form
- Add check/uncheck all links for each size on the form (woo JS magic)

### CODE REVIEW

File: app/assets/javascripts/application.js [hunk 1]
```diff
@@ -21,3 +21,18 @@ $(function(){ $(document).foundation(); });
 $(document).on('change', '.autosubmitme', function() {
   $(this).parents('form').submit();
 });
+
+$(document).on('click', '.suites-fields .check-all', function() {
+  set_checkbox_of_parent(this, '.suites-fields', true);
+  return false; // prevent scrolling behavior
```
Comment (orenyk) [inline app/assets/javascripts/application.js]: Apparently returning false in the other function doesn't carry through to the callback (as I would expect in Ruby).

File: app/models/draw_suites_update.rb [hunk 2]
```diff
@@ -4,7 +4,7 @@
 class DrawSuitesUpdate
   include ActiveModel::Model
 
-  attr_reader :size, :suite_ids, :drawn_suite_ids, :undrawn_suite_ids
+  attr_reader :suite_ids, :drawn_ids, :drawless_ids
```
Comment (orenyk) [inline app/models/draw_suites_update.rb]: I'm actually not sure we even need these readers at this point.

File: app/models/draw_suites_update.rb [hunk 3]
```diff
@@ -105,4 +104,27 @@ def error(error)
...
+    @all_suite_sizes ||= SuiteSizesQuery.new(Suite.available).call
+  end
+
+  def params_for(size)
+    param_headings = %w(suite_ids drawn_ids drawless_ids)
+    param_headings.map { |p| "#{p}_#{size}".to_sym }
+  end
+
+  def respond_to_missing?(method_name, include_all = false)
+    valid_method_names.include?(method_name) || super
```
Comment (esoterik) [inline app/models/draw_suites_update.rb]: what does `super` hit here?

File: app/controllers/draws_controller.rb [hunk 4]
```diff
@@ -243,4 +249,33 @@ def process_student_assignment
...
+
+  def suite_edit_param_headings
+    %w(suite_ids drawn_ids drawless_ids)
+  end
+
+  # This method is necessary to consolidate the suite update params from all the
+  # different suite sizes into three distinct params - this could be moved to
+  # the DrawSuitesUpdate service object in the future since we're already
+  # setting up getters for all of the possible param instance variables, but
+  # this could just as easily live in the controller.
```
Comment (esoterik) [inline app/controllers/draws_controller.rb]: I think this should go in the service object

File: app/controllers/draws_controller.rb [hunk 5]
```diff
@@ -199,14 +198,21 @@ def calculate_ungrouped_students_metrics
     @ungrouped_students.delete('off_campus')
   end
 
-  def prepare_suites_edit_data # rubocop:disable AbcSize
+  def prepare_suites_edit_data # rubocop:disable AbcSize, MethodLength
```
Comment (esoterik) [inline app/controllers/draws_controller.rb]: this makes me feel like we need a whole separate `DrawSuitesController` but leaving it alone for now

Comment (orenyk) [inline app/controllers/draws_controller.rb]: We definitely do.

File: app/models/draw_suites_update.rb (hunk 3 above)
Comment (orenyk) [inline app/models/draw_suites_update.rb]: It hits the default behavior in `Object` I believe, which is `false` unless it shows up in the Ruby method inheritance chain.

File: app/controllers/draws_controller.rb (hunk 4 above)
Comment (orenyk) [inline app/controllers/draws_controller.rb]: Ok, I'll move it and request another review.

**Comment by orenyk** [comment]
@esoterik back to you!

**Comment by orenyk** [comment]
Oh sorry forgot about that change, updating now.

**Comment by orenyk** [comment]
Also rebasing on #99 since that will be next to merge, I think.

**Comment by orenyk** [comment]
@esoterik back to you, just verify that you like how it looks / works and we should be good to merge.

**Pull request merged**